llama_stack/distribution/routers/vector_io.py

```diff
@@ -76,6 +77,42 @@ async def _get_first_embedding_model(self) -> tuple[str, int] | None:
             logger.error(f"Error getting embedding models: {e}")
             return None
 
+    async def _resolve_embedding_model(self, explicit_model: str | None = None) -> tuple[str, int]:
+        """Apply precedence rules to decide which embedding model to use.
+
+        1. If *explicit_model* is provided, verify dimension (if possible) and use it.
+        2. Else use the global default in ``vector_store_config``.
+        3. Else raise ``MissingEmbeddingModelError``.
+        """
+
+        # 1. explicit override
+        if explicit_model is not None:
+            # We still need a dimension; try to look it up in routing table
+            all_models = await self.routing_table.get_all_with_type("model")
+            for m in all_models:
+                if getattr(m, "identifier", None) == explicit_model:
+                    dim = m.metadata.get("embedding_dimension")
+                    if dim is None:
+                        raise ValueError(
+                            f"Failed to use embedding model {explicit_model}: found but has no embedding_dimension metadata"
+                        )
+                    return explicit_model, dim
+            # If not found, dimension unknown - defer to caller
+            return explicit_model, None  # type: ignore
+
+        # 2. global default
+        cfg = VectorStoreConfig()  # picks up env vars automatically
+        if cfg.default_embedding_model is not None:
+            return cfg.default_embedding_model, cfg.default_embedding_dimension or 384
+
+        # 3. error - no default
+        class MissingEmbeddingModelError(RuntimeError):
+            pass
```

Comment:
I wonder if there is somewhere else we might want to store new custom Exception classes?